### ISSUE TITLE
Should we add a `VM#setCustomMetadata`?

### ISSUE THREAD

**Issue by stephenplusplus** [googleapis/nodejs-compute#170]
This came up when a PR from @pickypg sparked [this conversation](https://github.com/googleapis/nodejs-compute/pull/169#discussion_r222037634).

Is there something we could do to make these concepts more distinct?:

- Our term "metadata", which is meant to represent the raw API resource
- The "metadata" property which could exist *inside* the raw API resource

Here's a related issue that could suggest we need to take this to the top ([nodejs-common](http://github.com/googleapis/nodejs-common), of course) to fully resolve: https://github.com/googleapis/nodejs-storage/issues/222

**Comment by pickypg** [comment]
TL;DR:

I think there's a few things that could have steered me down the right path:

- Providing a `VM#setCustomMetadata` method would differentiate from `VM#getMetadata` enough to tip me off that there's a difference.
   - Ideally a future major version (1.0 since it's only alpha now) would remove `VM#setMetadata`.
- Providing a corresponding `VM#getCustomMetadata` helper method with similar documentation.
- The documentation could make it clearer that `/setMetadata` (and its corresponding APIs in other libraries / REST) is for custom metadata only, and note the more generic usage of "metadata" while making that distinction.
    - As long as it exists, this should apply to `VM#setMetadata` as well.

-----

As for background of how I got here, I started from the merged PR's comment:

> Overall, it's hard to decide if it's a documentation problem or a code problem. And it could just be that neither are solely responsible, but both could be improved.

I think it is a bit of both as you suggest. My first usage of GCP was through the Compute Engine's UI without any other past usage of GCP (although plenty relative to AWS and Azure). The terminology between each Cloud Provider is naturally unique, but they all share a lot of similarities beyond terminology. I ended up exploring a lot of the Compute Engine via the APIs, without reading the documentation and I became used to metadata being what you describe as _custom_ metadata.

From there, I had a need to set some custom metadata dynamically and so I started to read the documentation for how to do that. As a result, whenever I read "metadata" I read it as I had seen it in the UI: custom metadata. Eventually I made it to this library where I found `VM#setMetadata` and proceeded to make it work. This rabbit hole led me to figure out that the current behavior of `VM#setMetadata` is to replace all custom metadata, rather than patch it, and I then went backwards to figure out how to fetch the metadata.

`VM#getMetadata` was the obvious solution for fetching it, but it became _very_ confusing once I did that because of the now-obvious issue: metadata is not custom metadata. That they are fundamentally different concepts sharing the same name is ultimately too confusing.

As an aside, it also didn't help that the custom metadata is returned differently (an `Array` of key/value paired `Object`s) than it is expected to be passed to the `VM#setMetadata` method (an `Object`), but that was the least of my issues.

**Comment by sofisl** [comment]
Hi, please refer to the new library, ^3.0.0.
